frontend/discourse/app/components/user-autocomplete-results.gjs

```diff
@@ -0,0 +1,128 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import { and, eq, not } from "truth-helpers";
+import avatar from "discourse/helpers/avatar";
+import icon from "discourse/helpers/d-icon";
+import { safeInvoke } from "discourse/lib/function-utils";
+import { formatUsername } from "discourse/lib/utilities";
+import scrollIntoView from "discourse/modifiers/scroll-into-view";
+
+/**
+ * Component for rendering user autocomplete results for the DAutocomplete modifier.
+ *
+ * This component handles rendering of users, emails, and groups in the autocomplete
+ * dropdown, and is designed to be used with DAutocomplete's `component` API.
+ *
+ * @component UserAutocompleteResults
+ * @param {Array} results - Array of autocomplete results (users, emails, groups)
+ * @param {number} selectedIndex - Currently selected index in the results list
+ * @param {Function} onSelect - Callback function triggered when a result is selected
+ * @param {Function} onRender - Optional callback function triggered after component renders
+ */
+export default class UserAutocompleteResults extends Component {
+  static TRIGGER_KEY = "@";
+
+  @tracked isInitialRender = true;
+
+  @action
+  handleResultClick(result, index, event) {
+    event.preventDefault();
+    event.stopPropagation();
+    safeInvoke(this.args.onSelect, result, index, event);
+  }
+
+  @action
+  handleInsert() {
+    safeInvoke(this.args.onRender, this.args.results);
+  }
+
+  @action
+  handleUpdate() {
+    this.isInitialRender = false;
+    safeInvoke(this.args.onRender, this.args.results);
+  }
+
+  <template>
+    <div
+      class="autocomplete ac-user"
+      {{didInsert this.handleInsert}}
+      {{didUpdate this.handleUpdate @selectedIndex}}
+    >
+      <ul>
+        {{#each @results as |result index|}}
+          {{#if result.isUser}}
+            <li
+              data-index={{result.index}}
+              {{scrollIntoView
+                (and (not this.isInitialRender) (eq index @selectedIndex))
+              }}
```

Comment:
This is quite complex, especially since it has to be repeated 3 times. How about adding a local modifier like

```js
maybeScrollIntoView = modifier((element) => {
  if(!this.alreadyScrolledIntoView){
    this.alreadyScrolledIntoView = true;
    element.scrollIntoView({block: "nearest"});
  }
});
```

And then it would be referenced in the template like

```hbs
<li {{this.maybeScrollIntoView}}>
```